.github/workflows/build_all_targets.yml

```
@@ -80,32 +87,57 @@ jobs:
     container:
       image: ${{ matrix.container }}
     steps:
+      - uses: runs-on/action@v2
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Git ownership workaround
         run: git config --system --add safe.directory '*'
 
-      - name: Setup ccache
-        uses: actions/cache@v4
+      - name: Setup ccache key elements
+        id: ccache-key
+        shell: bash
+        run: |
+          # create a checksum of the container tag to use as part of the ccache key
+          # this ensures that if the container changes we do not use an old cache
+          # this is important as different containers may have different compiler versions
+          # Example inputs:
+          # Container: ghcr.io/px4/px4-dev:v1.16.0-rc1-258-g0369abd556
+          # Container SHA256: 359e7171b82888bb60fea37edded382063ab7132aa1adbdc8d978ff4cf89a252
+          container_checksum=$(sha256sum <<< "${{ matrix.container }}" | cut -d' ' -f1)
+          echo "container_checksum=$container_checksum" >> $GITHUB_OUTPUT
+
+      # ccache key breakdown:
+      # ccache-<system os>-<system arch>-<container tag checksum>-<builder group>-
+      # ccache-<linux>-<arm64>-<359e7171b...>-<aarch64-0>-
+      # ccache-<linux>-<x64>-<359e7171b...>-<nuttx-0>-
+      - name: Restore ccache from key
+        id: cc_restore
+        uses: actions/cache/restore@v4
         with:
           path: ~/.ccache
-          key: ${{ matrix.group }}-ccache-${{ needs.group_targets.outputs.timestamp }}
-          restore-keys: ${{ matrix.group }}-ccache-${{ needs.group_targets.outputs.timestamp }}
-
-      - name: Configure ccache
+          key: ${{ format('ccache-{0}-{1}-{2}-{3}', runner.os, matrix.runner, steps.ccache-key.outputs.container_checksum, matrix.group) }}
+          restore-keys: |
+            ccache-${{ runner.os }}-${{ matrix.runner }}-${{ steps.ccache-key.outputs.container_checksum }}-${{ matrix.group }}-
+            ccache-${{ runner.os }}-${{ matrix.runner }}-${{ steps.ccache-key.outputs.container_checksum  }}-
+            ccache-${{ runner.os }}-${{ matrix.runner }}-
+            ccache-${{ runner.os }}-
+            ccache-
+
+      - name: Set ccache defaults and show stats
         run: |
             mkdir -p ~/.ccache
             echo "base_dir = ${GITHUB_WORKSPACE}" > ~/.ccache/ccache.conf
             echo "compression = true" >> ~/.ccache/ccache.conf
             echo "compression_level = 6" >> ~/.ccache/ccache.conf
-            echo "max_size = 120M" >> ~/.ccache/ccache.conf
+            echo "max_size = 1G" >> ~/.ccache/ccache.conf
```

Let's keep this as is and monitor the ccache stats going forward.

```suggestion
            echo "max_size = 120M" >> ~/.ccache/ccache.conf
```